src/editors/containers/VideoUploadEditor/hooks.js

```diff
@@ -0,0 +1,42 @@
+import * as requests from '../../data/redux/thunkActions/requests';
+
+export const uploadVideo = ({ dispatch, supportedFiles }) => {
+  const data = { files: [] };
+  supportedFiles.forEach((file) => {
+    data.files.push({
+      file_name: file.name,
+      content_type: file.type,
+    });
+  });
+  dispatch(requests.uploadVideo({
+    data,
+    onSuccess: (response) => {
+      const { files } = response.json();
+      Object.values(files).forEach((fileObj) => {
```

Comment:
@pkulkark This will be used when a user wants to add a single video to a course, so it shouldn't allow multiple file uploads.

For context, the user flow will be:
1. From studio, a user wants to add a video to a course.
2. User directly clicks on the video block in a unit, and is taken to a gallery page to browse available videos (being worked on a separate task).
3. Uses decides to upload a new video from that gallery, click the upload button and is taken to the page you are implementing here.
4. He either uploads a **single video** or adds a youtube URL.
5. Once the upload is complete, the user is taken to the video edit page with the information pre-filled there.

For your task, the only relevant item is (4) - building the upload page.

Reply:
Ack. Will remove the flag and update the logic.

Reply:
@pkulkark Do you think this function should be simplified to handle only one file upload?

Reply:
@Cup0fCoffee The response here is from the edx-platform API, which returns a list of files, as the API supports multiple file uploads. See https://github.com/openedx/edx-platform/blob/master/cms/djangoapps/contentstore/views/videos.py#L724-L742. Although in this case, it would return a list containing a single file, I left it as is in the interest of time (also didn't think it would affect computationally - let me know if I'm wrong about that).

Reply:
I haven't realized that. Thank you for clarifying. In that case I think it's ok.